src/engine/IdTable.h

```diff
@@ -109,12 +123,22 @@ class IdTableImpl {
       return *reinterpret_cast<const value_type*>(_data + (_row * COLS));
     }
 
+    // This is indeed the correct way to implement such things
+    pointer operator->() {
+      return reinterpret_cast<value_type*>(_data + (_row * COLS));
+    }
+
+    const value_type* operator->() const {
```

Comment:
As below, can you explain why the non-const version returns `pointer` while the const version returns `const value_type*` when that should be the same as `const pointer` according to the definition of `value_type` and `pointer`

Reply:
Same reason, but we could get rid of the casting here also.